src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -787,6 +789,49 @@ void primitive_inst::do_runtime_skip_reorder() {
     }
 }
 
+void primitive_inst::do_runtime_skip_gather() {
+    // Target pattter:
+    // [kv_cache_input / beam_idx_input] => Gather => Concat
+    auto is_optimizable_gather = [&] {
```

Comment:
I think it makes sense to check that gather doesn't have any fused ops or output paddings

Reply:
Fixed